builtin/credential/cert/path_login.go

```diff
@@ -371,6 +382,33 @@ func (b *backend) checkForValidChain(chains [][]*x509.Certificate) bool {
 	return false
 }
 
+func (b *backend) checkCIDR(cert *CertEntry, req *logical.Request) error {
```

Comment:
Considering that this function can be potentially used by all the backends that introduces `bound_cidr` option, we might want to pull this out as a helper, possibly in `helper/cidrutil` package? It would then also make sense to accept `([]*sockaddr.SockAddrMarshaler, *logical.Connection)` as parameters instead of `(*CertEntry, *logical.Request)`.

Reply:
Sure, I can do that. I was thinking I'd do it in the next iteration because it's not yet reused in this PR, but I can just do it now too.